### ISSUE TITLE
Move window with diffrent dpi monitor

### ISSUE THREAD

**Issue by Amroll** [micdenny/WpfScreenHelper#7]
Hi!
Thanks for giving a so nice lib for multi-screen.
There is a mistake and the step is:
1. Two moniter. The primary resolution is 3840,2160 and dpi is 300%, the other resolution is 1920,1080 and dpi is 100%
2. When I run WpfScreenHelper.DpiTestWpfApp.exe at the primary screen, It works ok. But when I selected the second screen and click Position "Any",the window is disappear.
3. Then I changed WpfScreenHelper.DpiTestWpfApp.exe code with use `WindowHelper.SetWindowPosition`.The situation is the same.

**Comment by micdenny** [comment]
@FoxPride can you help on this because I think is because of #5 am I wrong?

**Comment by FoxPride** [comment]
@Amroll can you provide additional info about end coordinates of window after moving it with code? May be it is because of windows size, try to change it and test a bit.

**Comment by Amroll** [comment]
@FoxPride  Sorry for reply so late.  
1. As my use for moving window,effective code 

```cs 
var otherscreen = Screen.AllScreens.First(item => item.DeviceName==#secondname#);
var myscreen= Screen.AllScreens.First(item => item.DeviceName != #secondname#);

double left = 0;
if (otherscreen .Bounds.Left >= myscreen.Bounds.Right)
    left = myscreen.Bounds.Right + 1; //the screen order is "myscreen" "otherscreen"
else if (otherscreen .Bounds.Right <= myscreen.Bounds.Left)
    left = myscreen.Bounds.Left - otherscreen .Bounds.Width + 10; //the screen order is "otherscreen" "myscreen"

this.Top = 0;
this.Left = left;   //"this" is window,and set it's Left ="left" ,so that can move to second window.
// as do upper
this.Dispatcher.Invoke(() => { this.Left= otherscreen.Bounds.X}); //now can use otherscreen's Bounds
```

2. The condition for use ```WindowHelper.SetWindowPosition(this Window window, WindowPositions pos, Rect bounds)```
```
- 3840,2160 screen's bounds is "0,0,1280,720", the other bounds is "3840,0,1920,1080"
- When select "\\.\DISPLAY2" and click "Position TopLeft" and the result for window.left is 3840. 
- Test: when I limit the argument "bounds" 's X between 1281 to 3199. It works.
```
So I think when move the window to other screen, the X or Y  must according to the current screen and can't out of the second screen bounds which is also according to the current screen.

BTW. I am not very good at English.I don't know if I can make myself clear.

**Comment by FoxPride** [comment]
@Amroll you have wrong coordinates for second monitor (it's in pixel format).
First screen is 3840x2160 with 300% scale. Bounds should be "0, 0, 1280, 720", PixelBounds equals "0, 0, 3840, 2160"
Second screen is 1920x1080 with 100%. Bounds should be "1280, 0, 1920, 1080", PixelBounds equals "3840, 0, 1920, 1080"

**Comment by Amroll** [comment]
@FoxPride Here are the debug info:
![firstscreen](https://user-images.githubusercontent.com/2509612/146103716-9c749963-55aa-40db-a283-341d777f325c.png)

![secondscreen](https://user-images.githubusercontent.com/2509612/146103707-78ce1be0-5607-4eea-8eb5-05e165a96ed2.png)
a interesting thing is when move window from second screen it must use the second screen's PixelBound.X which is 3840.
for examle: `this.Left=secondscreen.Bounds(current is 3840).X-this.ActualWidth.` .But when window show on firstscreen,it can't detect the right DPI scale(also use the secondscreen's dpi).  maybe is the WPF's BUG.

**Comment by FoxPride** [comment]
@Amroll do you have any manifest settings for your app activated [see](https://github.com/micdenny/WpfScreenHelper/blob/master/test/WpfScreenHelper.DpiTestWpfApp/app.manifest)? And what framework you are using for your app? (Checkout this [page](https://github.com/Microsoft/WPF-Samples/tree/master/PerMonitorDPI))

**Comment by Amroll** [comment]
@FoxPride I just debug the `WpfScreenHelper.DpiTestWpfApp` which in the WpfScreenHelper solution

**Comment by FoxPride** [comment]
@Amroll now I see problem in code logic but I need to do some tests for final confirm. I'll look into it and post update (hopefully this month).

**Comment by Amroll** [comment]
@FoxPride OK, Thanks.  I can do some test if you need.

**Comment by enginkirmaci** [comment]
@Amroll I think this problem is related to how window handled screen changes by system. try calling WindowHelper.SetWindowPosition twice if you want to move window to another screen that has different dpi.

**Comment by micdenny** [comment]
closed by #10

**Comment by micdenny** [comment]
[v2.0.0 has been released](https://www.nuget.org/packages/WpfScreenHelper/2.0.0)
